src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2109,12 +2109,18 @@ private unsafe ParsingError PrivateParseMinimal()
                             _flags |= Flags.UncPath;
                             idx = i;
                         }
+                        else if (!IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && i - idx == 3 && pUriString[i - 1] == '/')
+                        {
```

Comment:
@tmds the UNC syntax is shown only for LocalPath, the absoluteUri value for the file scheme, should not have converted the backslash to forward slash. Can you please make the change to apply for all file:// uris on Unix, irrespective of hostname. And please add tests to test these cases and expected values for AbsoluteUri, since LocalPath is not the only metric. Once you update the PR, we can work from there, to finalize the behavior.